src/client/Microsoft.Identity.Client/ManagedIdentity/V2/ImdsV2ManagedIdentitySource.cs

```diff
@@ -194,9 +189,90 @@ public static AbstractManagedIdentity Create(RequestContext requestContext)
+        {
+            var queryParams = ImdsV2QueryParamsHelper(_requestContext);
+
+            // TODO: add bypass_cache query param in case of token revocation. Boolean: true/false
+
+            var headers = new Dictionary<string, string>
+            {
+                { "Metadata", "true" },
+                { "x-ms-client-request-id", _requestContext.CorrelationId.ToString() }
+            };
+            
+            var body = $"{{\"csr\":\"{csr}\"}}";
+
+            IRetryPolicyFactory retryPolicyFactory = _requestContext.ServiceBundle.Config.RetryPolicyFactory;
+            IRetryPolicy retryPolicy = retryPolicyFactory.GetRetryPolicy(RequestType.Imds);
+
+            HttpResponse response = null;
+
+            try
+            {
+                response = await _requestContext.ServiceBundle.HttpManager.SendRequestAsync(
+                    ImdsManagedIdentitySource.GetValidatedEndpoint(_requestContext.Logger, CertificateRequestPath, queryParams),
+                    headers,
+                    body: new StringContent(body, System.Text.Encoding.UTF8, "application/json"),
+                    method: HttpMethod.Post,
+                    logger: _requestContext.Logger,
+                    doNotThrow: false,
+                    mtlsCertificate: null,
+                    validateServerCertificate: null,
+                    cancellationToken: _requestContext.UserCancellationToken,
+                    retryPolicy: retryPolicy)
+                .ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                throw MsalServiceExceptionFactory.CreateManagedIdentityException(
+                    MsalError.ManagedIdentityRequestFailed,
+                    $"[ImdsV2] ImdsV2ManagedIdentitySource.ExecuteCertificateRequestAsync failed.",
+                    ex,
+                    ManagedIdentitySource.ImdsV2,
+                    (int)response.StatusCode);
+            }
+
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                throw MsalServiceExceptionFactory.CreateManagedIdentityException(
+                    MsalError.ManagedIdentityRequestFailed,
+                    $"[ImdsV2] ImdsV2ManagedIdentitySource.ExecuteCertificateRequestAsync failed due to HTTP error. Status code: {response.StatusCode} Body: {response.Body}",
+                    null,
+                    ManagedIdentitySource.ImdsV2,
+                    (int)response.StatusCode);
+            }
+
+            var certificateRequestResponse = JsonHelper.DeserializeFromJson<CertificateRequestResponse>(response.Body);
+            CertificateRequestResponse.Validate(certificateRequestResponse);
+
+            return certificateRequestResponse;
+        }
+
         protected override ManagedIdentityRequest CreateRequest(string resource)
         {
+            var csrMetadata = GetCsrMetadataAsync(_requestContext, false).GetAwaiter().GetResult();
```

Comment:
please consider refactoring this part to use async/await.

Reply:
I can't without changing the base method, which is used by every other MI source

Reply:
+1 to this. No sync over async.

Reply:
per offline discussion, we're changing the CreateRequest signature to an async mthod